lib/tether/ops_linux.go

```diff
@@ -358,6 +358,27 @@ func linkAddrUpdate(old, new *net.IPNet, t Netlink, link netlink.Link) error {
 	return nil
 }
 
+func updateRoutes(t Netlink, link netlink.Link, endpoint *NetworkEndpoint) error {
+	gw := endpoint.Network.Gateway
+	if ip.IsUnspecifiedIP(gw.IP) {
+		return nil
+	}
+
+	for _, d := range endpoint.Network.Destinations {
+		r := &netlink.Route{
+			LinkIndex: link.Attrs().Index,
+			Dst:       &d,
+			Gw:        gw.IP,
+		}
+
+		if err := t.RouteAdd(r); err != nil && !os.IsNotExist(err) {
+			log.Errorf("failed to add route for destination %s via gateway %s", d, gw.IP)
+		}
+	}
+
+	return updateDefaultRoute(t, link, endpoint)
```

Comment:
Its called for all networks, but the first thing that function checks if its the "default" network. Effectively it will only do something for the external network since its the only one marked as default in the config.